tests/templates/test_subroutines/test_qsvt.py

```diff
@@ -0,0 +1,471 @@
+                0.3,
+                [-0.8104500678299933, 1.520692517929803, 0.7603462589648997],
+                [0],
+                0.009,
+            ),  # angles from pyqsp give 0.1*x**2
+        ],
+    )
+    def test_output_wx(self, A, phis, wires, result):
+        """Test that qml.qsvt produces the correct output."""
+        dev = qml.device("default.qubit", wires=len(wires))
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.qsvt(A, phis, wires, convention="Wx")
+            return qml.expval(qml.PauliZ(wires=0))
+
+        assert np.isclose(np.real(qml.matrix(circuit)())[0][0], result, rtol=1e-3)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize(
+        ("input_matrix", "angles", "wires"),
+        [([[0.1, 0.2], [0.3, 0.4]], [0.1, 0.2], [0, 1])],
+    )
+    def test_qsvt_torch(self, input_matrix, angles, wires):
+        """Test that the qsvt function matrix is correct for torch."""
+        import torch
+
+        default_matrix = qml.matrix(qml.qsvt(input_matrix, angles, wires))
+
+        input_matrix = torch.tensor(input_matrix, dtype=float)
+        angles = torch.tensor(angles, dtype=float)
+
+        op = qml.qsvt(input_matrix, angles, wires)
+
+        assert np.allclose(qml.matrix(op), default_matrix)
+        assert qml.math.get_interface(qml.matrix(op)) == "torch"
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize(
+        ("input_matrix", "angles", "wires"),
+        [([[0.1, 0.2], [0.3, 0.4]], [0.1, 0.2], [0, 1])],
+    )
+    def test_qsvt_jax(self, input_matrix, angles, wires):
+        """Test that the qsvt function matrix is correct for jax."""
+        import jax.numpy as jnp
+
+        default_matrix = qml.matrix(qml.qsvt(input_matrix, angles, wires))
+
+        input_matrix = jnp.array(input_matrix)
+        angles = jnp.array(angles)
+
+        op = qml.qsvt(input_matrix, angles, wires)
+
+        assert np.allclose(qml.matrix(op), default_matrix)
+        assert qml.math.get_interface(qml.matrix(op)) == "jax"
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize(
+        ("input_matrix", "angles", "wires"),
+        [([[0.1, 0.2], [0.3, 0.4]], [0.1, 0.2], [0, 1])],
+    )
+    def test_qsvt_tensorflow(self, input_matrix, angles, wires):
+        """Test that the qsvt function matrix is correct for tensorflow."""
+        import tensorflow as tf
+
+        default_matrix = qml.matrix(qml.qsvt(input_matrix, angles, wires))
+
+        input_matrix = tf.Variable(input_matrix)
+        angles = tf.Variable(angles)
+
+        op = qml.qsvt(input_matrix, angles, wires)
+
+        assert np.allclose(qml.matrix(op), default_matrix)
+        assert qml.math.get_interface(qml.matrix(op)) == "tensorflow"
+
+    @pytest.mark.parametrize(
+        ("A", "phis", "wires"),
+        [
+            (
+                [[0.1, 0.2], [0.3, 0.4]],
+                [0.1, 0.2, 0.3],
+                [0, 1],
+            )
+        ],
+    )
+    def test_qsvt_grad(self, A, phis, wires):
```

Comment:
Are we confident about the gradient working in other interfaces?

Reply:
They work for some limited cases that we tested. But we plan to add more thorough testing in another PR.